jerry-core/ecma/base/ecma-module.c

```diff
@@ -823,6 +885,46 @@ ecma_module_connect_imports (void)
   return ECMA_VALUE_EMPTY;
 } /* ecma_module_connect_imports */
 
+/**
+ * Checks if indirect exports in the current context are resolvable.
+ * Note: See 15.2.1.16.4 / 9.
+ *
+ * @return ECMA_VALUE_ERROR - if an error occured
+ *         ECMA_VALUE_EMPTY - otherwise
+ */
+static ecma_value_t
+ecma_module_check_indirect_exports (ecma_module_t *module_p)
+{
+  ecma_module_node_t *indirect_export_p = module_p->indirect_exports_p;
+  while (indirect_export_p != NULL)
+  {
+    ecma_module_names_t *name_p = indirect_export_p->module_names_p;
+    while (name_p != NULL)
+    {
+      ecma_module_record_t record;
+      ecma_value_t result = ecma_module_resolve_export (indirect_export_p->module_request_p,
+                                                        name_p->local_name_p,
+                                                        &record);
+
+      if (ECMA_IS_VALUE_ERROR (result))
+      {
+        return result;
+      }
+
+      if (record.module_p == NULL)
```

Comment:
At this point the `result` is an empty value? If so we could add an assert on it.